beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -366,10 +374,27 @@ where
                 substream: out,
                 request,
             };
-            if let Some(_) = self
-                .outbound_substreams
-                .insert(id, (awaiting_stream, delay_key, protocol))
-            {
+            let response_chunk_count: Option<u64> = match protocol {
```

Comment:
I think this could be simplified to match just the request. Something like:
```rust
let response_chunk_count = match request {
    RPCRequest::BlocksByRange(req) => Some(req.cont),
    RPCRequest::BlocksByRoot(req) => Some(req.block_roots.len() as u64),
    _ => None // other requests don't have a predetermined size
}
```